R/get_national_data.R

```diff
@@ -0,0 +1,155 @@
+#' Get national-level data for countries globally, sourced from the ECDC or WHO.
+#' 
+#' @description Gets raw data using the source-specific function. Includes ISO country codes. 
+#' Then adds columns which were missing from the raw data (calculating cumulative counts from new dailies and vice versa),
+#' cleans and sanitises further. Adds rows and columns of NA values so that data is in a standard format. 
+#' 
+#' @param country Character A string specifying the country to get data from. Not case dependent. Defaults to all countries.
+#' Name should be the English name.
+#' @param totals Boolean. If TRUE, returns totalled data per country up to today's date. If FALSE, returns the full dataset stratified by date and country.
+#' @param source Character A string specifying the data source: "WHO", or "ECDC". Not case dependent. Defaults to ECDC.
+#' @return A tibble with data related to cases, deaths, hospitalisations, recoveries and testing.
+#' @importFrom dplyr %>% group_by arrange select ungroup do everything
+#' @importFrom tidyr drop_na fill
+#' @importFrom tibble tibble
+#' @export
+#' @examples
+#' 
+#' \dontrun{
+#'  get_national_data(country = "canada", totals = TRUE, source = "WHO")
+#' }
+#' 
+
+get_national_data <- function(country = NULL, totals = FALSE, source = "ecdc"){
+  
+  # Error handling -------------------------------------------------------------------
+  if (!is.null(country) & !(is.character(country))){
+    stop("The country variable should be a character variable.")
+  }
+  
+  if (!(is.logical(totals))){
+    stop("The totals variable should be a logical (TRUE/FALSE) variable.")
+  }
+
+  if (!is.null(source) & !(is.character(source))){
+    stop("The source variable should be a character variable ('WHO' or 'ECDC').")
+  }
+  
+  
+ source <- tolower(source)
+ 
+# Match country name -----------------------------------------------------
+ 
+  input_country_name <- tolower(country) # Set to a different variable name to avoid mixups with "country" var in data
+  if(length(input_country_name) == 0){input_country_name <- NULL} # Reset to NULL to return all countries
+
+  if (!is.null(input_country_name)) {
+
+    name_match <- read.csv(here::here("data-raw", "country_names.csv"), colClasses = "character", na.strings = "")
```

Comment:
this will only work when using the package internally. Instead need to add the country_names.csv to the data of the package. To do this see the examples in EpiNow (the gist is to have a script in data-raw that reads in and passes to usethis::use_data and then add a documentation R file for the data set. Once this is done the data can be used by doing covidregionaldata::country_names

Reply:
if it were possible to have UN region here (or some other region) that would save some messy work down the line in  the website code (though that is already in place so no rush)

Reply:
- Replaced country_names.csv with an R package that does it better. 
- Added un region names as variable in output data